src/hooks/useModeratorTrackListener/useModeratorTrackListener.tsx

```diff
@@ -0,0 +1,51 @@
+import useVideoContext from '../useVideoContext/useVideoContext';
+import { useEffect } from 'react';
+import { RemoteParticipant, Track } from 'twilio-video';
+import { ParticipantIdentity } from '../../utils/participantIdentity';
+import { TRACK_TYPE } from 'utils/displayStrings';
+import isModerator from 'utils/rbac/roleChecker';
+import { IMuteRemoteParticipantMessage } from '../../utils/muteRemoteParticipantMessage';
+
+export default function useModeratorTrackListener() {
+  const { room, localTracks } = useVideoContext();
+
+  useEffect(() => {
+    const handleRemoteParticipant = (participant: RemoteParticipant) => {
+      if (isModerator(ParticipantIdentity.Parse(participant.identity).partyType)) {
+        participant.on('trackSubscribed', track => {
+          console.log(`Participant "${participant.identity}" added ${track.kind} Track ${track.sid}`);
+          if (track.kind === 'data') {
+            track.on('message', data => {
+              console.log('message recieved ' + data);
+              handleMessage(data);
+            });
+          }
+        });
+      }
+    };
+
+    function handleMessage(data: any) {
```

Comment:
please add the layer here that will serve in the future to distinguish different types of messages and route them

Reply:
Doing as part of CA-443